pkg/controller/argocd/util.go

```diff
@@ -655,6 +659,67 @@ func (r *ReconcileArgoCD) reconcileResources(cr *argoprojv1a1.ArgoCD) error {
 	return nil
 }
 
+func (r *ReconcileArgoCD) deleteClusterResources(cr *argoprojv1a1.ArgoCD) error {
+	selector, err := argocdInstanceSelector(cr.Name)
+	if err != nil {
+		return err
```

Comment:
See above, for the note about the error.

Something to consider, if the result of `filterObjectsBySelector` returned a useful enough error, then you could just return it, instead of rewrapping :-)

Reply:
Updated the error message